File: user-service-java/src/main/java/com/project/userService/services/RabbitNotificationServiceImpl.java
package com.project.userService.services;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
@RequiredArgsConstructor
@Slf4j
public class RabbitNotificationServiceImpl implements NotificationService {
    private final RabbitTemplate rabbitTemplate;

    @Override
    public <T> void sendNotification(String title, T object) {
        ObjectMapper mapper = new ObjectMapper();
        var values = mapper.convertValue(object, new TypeReference<Map<String, Object>>() {
        });
        values.put("type", "INFO");
        values.put("title", title);
        try {
            var message = mapper.writeValueAsString(values);
            rabbitTemplate.convertAndSend("service.notification", "notification-routing-key", message);
        } catch (Exception e) {
            log.error(e.getMessage());
        }
    }
}

File: notification-service-kotlin/src/main/kotlin/com/project/notificationService/service/RabbitNotificationServiceImpl.kt
package com.project.notificationService.service

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import com.project.notificationService.config.RabbitConfig
import com.project.notificationService.models.Notification
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.amqp.core.Message
import org.springframework.amqp.rabbit.annotation.EnableRabbit
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Service


@Service
@EnableRabbit
class RabbitNotificationServiceImpl(
    private val service: NotificationService,
    private val mapper: ObjectMapper
) {
    @RabbitListener(queues = [RabbitConfig.NOTIFICATION_SERVICE_QUEUE])
    fun receiveNotification(message: Message) {
        val body = String(message.body).removePrefix("\"").removeSuffix("\"").replace("\\", "")
        try {
            val notification: Notification = mapper.readValue<Notification>(body)
            service.saveNotification(notification)
        } catch (e: Exception) {
            log.error(e.message)
        }
    }

    companion object {
        private val log: Logger = LoggerFactory.getLogger(this::class.java)
    }
}